### ISSUE TITLE
Compile time error with Swift Package manager 

### ISSUE THREAD

**Issue by alex5733** [danielgindi/Charts#4533]
Hi I am getting multiple compile time error with Charts 4.0 with Xcode Version 12.1 (12A7403)

Charts/Renderers/AxisRendererBase.swift:16:12: Type 'AxisRendererBase' does not conform to protocol 'Renderer'
Charts/Renderers/Renderer.swift:19:9: Protocol requires property 'viewPortHandler' with type 'ViewPortHandler'
Charts/Renderers/AxisRendererBase.swift:15:2: Only classes that inherit from NSObject can be declared @objc
Charts/Renderers/AxisRendererBase.swift:26:9: 'super' members cannot be referenced in a root class
Charts/Renderers/AxisRendererBase.swift:68:56: Cannot find 'viewPortHandler' in scope
Charts/Renderers/AxisRendererBase.swift:70:100: Cannot find 'viewPortHandler' in scope
Charts/Renderers/AxisRendererBase.swift:71:100: Cannot find 'viewPortHandler' in scope
Charts/Renderers/AxisRendererBase.swift:109:36: Value of type 'Double' has no member 'roundedToNextSignficant'
Charts/Utils/ChartUtils.swift:160:66: Cannot find type 'NSTextAlignment' in scope
Charts/Renderers/ScatterChartRenderer.swift:174:50: Extra argument 'align' in call
Charts/Renderers/ScatterChartRenderer.swift:174:50: Cannot infer contextual base in reference to member 'center'
Charts/Renderers/XAxisRenderer.swift:252:25: Cannot find 'NSParagraphStyle' in scope
Charts/Renderers/XAxisRenderer.swift:456:20: Cannot find type 'NSTextAlignment' in scope
Charts/Renderers/XAxisRendererHorizontalBarChart.swift:257:28: Cannot find type 'NSTextAlignment' in scope
Charts/Renderers/YAxisRenderer.swift:126:20: Cannot find type 'NSTextAlignment' in scope
Charts/Renderers/YAxisRenderer.swift:47:24: Cannot find type 'NSTextAlignment' in scope
Charts/Renderers/YAxisRenderer.swift:303:24: Cannot find type 'NSTextAlignment' in scope
Charts/Renderers/YAxisRendererHorizontalBarChart.swift:138:38: Extra argument 'align' in call
Charts/Renderers/YAxisRendererHorizontalBarChart.swift:138:44: Missing argument for parameter 'angleRadians' in call
Charts/Utils/ChartUtils.swift:178:15: 'drawText(_:at:anchor:angleRadians:attributes:)' declared here
Charts/Renderers/YAxisRendererHorizontalBarChart.swift:138:38: Cannot infer contextual base in reference to member 'center'
Charts/Renderers/YAxisRendererHorizontalBarChart.swift:261:28: Cannot find type 'NSTextAlignment' in scope
Charts/Renderers/YAxisRendererRadarChart.swift:161:24: Extra argument 'align' in call
Charts/Renderers/YAxisRendererRadarChart.swift:161:33: Missing argument for parameter 'angleRadians' in call
Charts/Utils/ChartUtils.swift:178:15: 'drawText(_:at:anchor:angleRadians:attributes:)' declared here
Charts/Utils/ChartUtils.swift:230:68: Cannot find type 'NSTextAlignment' in scope
Charts/Renderers/LineChartRenderer.swift:588:50: Extra argument 'align' in call
Charts/Renderers/LineChartRenderer.swift:588:50: Cannot infer contextual base in reference to member 'center'
Charts/Renderers/PieChartRenderer.swift:476:49: Extra argument 'align' in call
Charts/Renderers/PieChartRenderer.swift:486:53: Extra argument 'align' in call
Charts/Renderers/PieChartRenderer.swift:499:53: Extra argument 'align' in call
Charts/Renderers/PieChartRenderer.swift:510:49: Extra argument 'align' in call
Charts/Renderers/PieChartRenderer.swift:527:50: Extra argument 'align' in call
Charts/Renderers/PieChartRenderer.swift:527:50: Cannot infer contextual base in reference to member 'center'
Charts/Renderers/PieChartRenderer.swift:535:54: Extra argument 'align' in call
Charts/Renderers/PieChartRenderer.swift:535:54: Cannot infer contextual base in reference to member 'center'
Charts/Renderers/PieChartRenderer.swift:547:54: Extra argument 'align' in call
Charts/Renderers/PieChartRenderer.swift:547:54: Cannot infer contextual base in reference to member 'center'
Charts/Renderers/PieChartRenderer.swift:557:50: Extra argument 'align' in call
Charts/Renderers/PieChartRenderer.swift:557:50: Cannot infer contextual base in reference to member 'center'
Charts/Renderers/RadarChartRenderer.swift:242:46: Extra argument 'align' in call
Charts/Renderers/RadarChartRenderer.swift:242:46: Cannot infer contextual base in reference to member 'center'
Charts/Renderers/BarChartRenderer.swift:674:75: Extra argument 'align' in call
Charts/Renderers/BarChartRenderer.swift:674:80: Missing argument for parameter 'angleRadians' in call
Charts/Utils/ChartUtils.swift:178:15: 'drawText(_:at:anchor:angleRadians:attributes:)' declared here
Charts/Renderers/BarChartRenderer.swift:679:75: Extra argument 'align' in call
Charts/Renderers/BubbleChartRenderer.swift:217:46: Extra argument 'align' in call
Charts/Renderers/BubbleChartRenderer.swift:217:46: Cannot infer contextual base in reference to member 'center'
Charts/Renderers/CandleStickChartRenderer.swift:331:50: Extra argument 'align' in call
Charts/Renderers/CandleStickChartRenderer.swift:331:50: Cannot infer contextual base in reference to member 'center'
Charts/Renderers/LegendRenderer.swift:551:66: Extra argument 'align' in call
Charts/Renderers/LegendRenderer.swift:551:70: Missing argument for parameter 'angleRadians' in call
Charts/Utils/ChartUtils.swift:178:15: 'drawText(_:at:anchor:angleRadians:attributes:)' declared here
Charts/Renderers/LegendRenderer.swift:551:66: Cannot infer contextual base in reference to member 'left'
Charts/Charts/ChartViewBase.swift:328:45: Extra argument 'align' in call
Charts/Charts/ChartViewBase.swift:328:54: Missing argument for parameter 'angleRadians' in call

**Comment by AjazDev** [comment]
Yes, facing same issue when trying to add Charts as SPM dependency.

**Comment by SumiMakito** [comment]
Same here.

**Comment by SumiMakito** [comment]
Same issue here.

I'm using Charts as a dependency in another Swift package.

**Comment by matt-dewitt** [comment]
Same issue here, doesn't compile when pulled in via SPM using Xcode 12.3

**Comment by matt-dewitt** [comment]
Bumped it down to version 3.6.0 and it seem ok now.

**Comment by radianttap** [comment]
A bunch of errors are thrown when using Charts 4.0.0 through SPM.

As @matt-dewitt said, going back to 3.6.0 passes compile. Now to see do features I need actually work. :)

**Comment by AjazDev** [comment]
I needed some of the features from the latest branch so I can't go back to 3.6.0.
For now I forked it and added "Import UIKit" in a bunch of files that were throwing the error... 🤨

**Comment by dudenamedjune** [comment]
@AjazDev Did that fix the errors?

**Comment by radianttap** [comment]
With changes from my #4539 PR, everything compiles fine.

**Comment by merichle** [comment]
I have same issue too in version 4.0.0

**Comment by rollinsio** [comment]
This appears to be back with the advent of XCode 13. However, when inspecting some of the changes that @radianttap made I suspect that this might actually be due to a regression.

**Comment by GrandFelix** [comment]
as @rollinsio mentioned. With xcode 13 this again becomes an issue.

**Comment by RobertBernardini** [comment]
The same thing is happening in our project when downloading the source files using Carthage. We use our own project settings to build the framework based on the source files rather than building the framework via Carthage. 

Considering that this has been reopened since October last year, has some priority been given to this to solve this issue so that we can get started using 4.0.0 again? We really wish to use the latest version with the new accessibility features which have been added since 3.6.0.
